presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -1404,6 +1413,31 @@ void PrestoServer::reportNodeStatus(proxygen::ResponseHandler* downstream) {
   http::sendOkResponse(downstream, json(fetchNodeStatus()));
 }
 
+void PrestoServer::handleGracefulShutdown(const std::vector<std::unique_ptr<folly::IOBuf>>& body, proxygen::ResponseHandler* downstream){
+    auto requestBody = body[0]->moveToFbString();
+    // Remove the trailing newline characters from the requestBody
+    requestBody.erase(requestBody.length()-2);
+    if (body.size()==1 && requestBody == "\"SHUTTING_DOWN\"") {
```

Comment:
`body.size() == 1` check should be before `body[0]->...` to handle empty request.
Include the newline in the string check and remove `requestBody.erase(requestBody.length()-2);`
```
if (body.size() == 1 && body[0]->toString() ==  "\"SHUTTING_DOWN\n\"") {

} else {

}
```

Reply:
Ok I have made the changes. Actually  "\n" is not needed it was a mistake on my side while sending the requests through postman I added a newline in the request body. Should I add code to trim the leading and trailing spaces or I should strictly expect "\"SHUTTING_DOWN\"" in the request body?

Reply:
> Should I add code to trim the leading and trailing spaces or I should strictly expect ""SHUTTING_DOWN"" in the request body?

Can you check what Java does?

Reply:
Java allows strings with leading and trailing spaces, that is why I was parsing to json, it was a clean way of removing these spaces. Should I go back to using parseJson() ?

Reply:
Let's use `folly::trimWhitespace` https://github.com/facebook/folly/blob/main/folly/String.h#L671 if you need to trim spaces.

Reply:
Done.